README.md

```diff
@@ -1,39 +1,124 @@
+| `-d, --debug` | `HIERARCHY_DEBUG` | `false` | Print debug output |
+| `--trace` | `HIERARCHY_TRACE` | `false` | Prints a diff after processing each file. This generates A LOT of output |
+| `-V, --version` | | | Print version and build information, then exit |
+
+### Merging
+
+The `hierarchy` utility processes the YAML structure as a deep merge, with the exception of lists. Lists are completely overwritten, and therefore, it is important to keep that in mind when using them.
+
+### Hierarchy
+
+The hierarchy is defined in the file `hierarchy.lst`. This is a simple text file that lists one include folder per line and supports comments prefixed with `#`. The directories listed can be relative or absolute (try to avoid) paths. You can have directories included that are higher or lower in the structure to control their precedence. You can look at examples [here](https://github.com/KohlsTechnology/hierarchy/blob/master/testdata/).
+
+#### Example content
+```
+../defaults    #this is the first ... lowest priority
+../marketing   #this is the second
+../development #this is the third ... highest priority
+```
+
+In this case, it will load all yaml files from `../defaults`, then merge it with everything in `../marketing`, and lastly merge it with everything in `../development`. You can also use the relative path `./`, which means that it will also load variables defined in contextDir directly (same folder level as `hierarchy.lst`). You can insert `./` in any desired order in the `hierarchy.lst`, thus determining its priority.
+
+#### Example
+
+Let's assume you have multiple applications that get deployed over different cloud providers. This application also has development, QA, and production environments. You can decide in whichever priority (order) the configuration files are merged.
+
+```
+defaults             # all applications will have this
+└── cloud            # settings specific to a cloud
+  └── environment    # settings specific to the environment level (e.g. development or production)
+    └── application  # settings specific to an application
+```
+
+In order to generate the final configuration for an application running in the development environment on cloud A, the `hierarchy.lst` could look like the below example.
+
+```
+# Hierarchy file for application "demo" running in "cloud A".
+# location `..../applications/demo/dev/clouda1.hierarchy.lst
+../../defaults
+../clouds/A
+./environments/dev
+./
+```
+
+### Environment variables in the hierarchy
+
+Hierarchy allows the use of operating system environment variables to make it even more flexible. The variables must be in the format `${NAME}`, only consist of letters, numbers, and underscores, and must start with a letter. The environment variable names will be converted to upper case to avoid ambiguity. If an environment variable is not found, the program will error out to avoid generating the wrong data.
```

Comment:
```suggestion
Hierarchy allows the use of environment variables to make it even more flexible. The variables must: be in the format `${NAME}`, only consist of letters, numbers, and underscores, and start with a letter. The environment variable names will be converted to upper case to avoid ambiguity. If an environment variable is not found, the program will error out to avoid generating the wrong data.
```